frontend/server/handlers/artifacts.ts

```diff
@@ -239,6 +254,91 @@ function getGCSArtifactHandler(options: { key: string; bucket: string }, peek: n
   };
 }
 
+function getVolumeArtifactsHandler(options: { bucket: string; key: string }, peek: number = 0) {
+  const { key, bucket } = options;
+  return async (req: Request, res: Response) => {
+    try {
+      const [pod, err] = await serverInfo.getHostPod();
+      if (err) {
+        res.status(500).send(err);
+        return;
+      }
+
+      if (!pod) {
+        res.status(500).send('Could not get server pod');
+        return;
+      }
+
+      // volumes not specified or volume named ${bucket} not specified
+      if (!Array.isArray(pod?.spec?.volumes) || !pod.spec.volumes.find(v => v?.name === bucket)) {
+        const message = `Failed to open volume://${bucket}/${key}, volume ${bucket} not found`;
```

Comment:
nit: `volume ${bucket} not found` -> `volume ${bucket} not mounted`, it's clearer how this volume should be found.